### PR TITLE
[Subversion] Add custom release data script

### PR DESCRIPTION
**Pull request by romainx**
Hello,

This is a custom data parser for [Apache Subversion](https://subversion.apache.org/) release data in the frame of endoflife-date/endoflife.date/pull/5567 implementing issue endoflife-date/endoflife.date/issues/5319.

Cheers

### CODE REVIEW

File: src/svn.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from bs4 import BeautifulSoup
+from common import dates, http, releasedata
+
+# https://regex101.com/r/CwvyT3/2 only non beta versions
+VERSION_PATTERN = re.compile(r"^(?:Subversion\s)(\d.\d+\.\d+$)")
+# https://regex101.com/r/GsimYd/1
+DATE_PATTERN = re.compile(r"^(?:\s\()(\w+,\s\d{1,2}\s\w+\s\d{4})")
+URL = "https://subversion.apache.org/docs/release-notes/release-history.html"
+
+with releasedata.ProductData("svn") as product_data:
```
Comment (marcwrobel) [inline src/svn.py:12]:
```suggestion
with releasedata.ProductData("apache-subversion") as product_data:
```

This and the script name must match the name of the markdown file in https://github.com/endoflife-date/endoflife.date/pull/5567.

Comment (marcwrobel) [reply src/svn.py]: Done.

File: src/apache-subversion.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import re
+
+from bs4 import BeautifulSoup
+from common import dates, http, releasedata
+
+# https://regex101.com/r/CwvyT3/2 only non beta versions
+VERSION_PATTERN = re.compile(r"^(?:Subversion\s)(\d.\d+\.\d+$)")
```
Comment (marcwrobel) [inline src/apache-subversion.py]:
```suggestion
VERSION_PATTERN = re.compile(r"^Subversion\s(\d.\d+\.\d+$)")
```

Remove unnecessary non-capturing group.

**Review by marcwrobel — APPROVED** [review]
Thanks for this contribution @romainx.

Tested, works as expected:
```
$ python src/apache-subversion.py
loaded data for apache-subversion from releases/apache-subversion.json
Fetching ['https://subversion.apache.org/docs/release-notes/release-history.html']
Fetched ['https://subversion.apache.org/docs/release-notes/release-history.html']
adding version 1.14.3 (2023-12-28 00:00:00+00:00) to apache-subversion
adding version 1.14.2 (2022-04-12 00:00:00+00:00) to apache-subversion
adding version 1.10.8 (2022-04-12 00:00:00+00:00) to apache-subversion
adding version 1.14.1 (2021-02-10 00:00:00+00:00) to apache-subversion
adding version 1.10.7 (2021-02-10 00:00:00+00:00) to apache-subversion
adding version 1.14.0 (2020-05-27 00:00:00+00:00) to apache-subversion
adding version 1.13.0 (2019-10-30 00:00:00+00:00) to apache-subversion
adding version 1.12.2 (2019-07-24 00:00:00+00:00) to apache-subversion
adding version 1.10.6 (2019-07-24 00:00:00+00:00) to apache-subversion
adding version 1.9.12 (2019-07-24 00:00:00+00:00) to apache-subversion
adding version 1.12.0 (2019-04-24 00:00:00+00:00) to apache-subversion
adding version 1.11.1 (2019-01-11 00:00:00+00:00) to apache-subversion
adding version 1.10.4 (2019-01-11 00:00:00+00:00) to apache-subversion
adding version 1.9.10 (2019-01-11 00:00:00+00:00) to apache-subversion
adding version 1.11.0 (2018-10-30 00:00:00+00:00) to apache-subversion
adding version 1.10.3 (2018-10-10 00:00:00+00:00) to apache-subversion
adding version 1.10.2 (2018-07-20 00:00:00+00:00) to apache-subversion
adding version 1.9.9 (2018-07-20 00:00:00+00:00) to apache-subversion
adding version 1.10.0 (2018-04-12 00:00:00+00:00) to apache-subversion
adding version 1.9.7 (2017-08-10 00:00:00+00:00) to apache-subversion
adding version 1.8.19 (2017-08-10 00:00:00+00:00) to apache-subversion
adding version 1.8.18 (2017-07-07 00:00:00+00:00) to apache-subversion
adding version 1.9.6 (2017-07-06 00:00:00+00:00) to apache-subversion
adding version 1.9.5 (2016-11-29 00:00:00+00:00) to apache-subversion
adding version 1.8.17 (2016-11-29 00:00:00+00:00) to apache-subversion
adding version 1.9.4 (2016-04-28 00:00:00+00:00) to apache-subversion
adding version 1.8.16 (2016-04-28 00:00:00+00:00) to apache-subversion
adding version 1.9.3 (2015-09-15 00:00:00+00:00) to apache-subversion
adding version 1.8.15 (2015-09-15 00:00:00+00:00) to apache-subversion
adding version 1.9.2 (2015-09-23 00:00:00+00:00) to apache-subversion
adding version 1.9.1 (2015-09-02 00:00:00+00:00) to apache-subversion
adding version 1.7.22 (2015-08-14 00:00:00+00:00) to apache-subversion
adding version 1.9.0 (2015-08-05 00:00:00+00:00) to apache-subversion
adding version 1.8.14 (2015-08-05 00:00:00+00:00) to apache-subversion
adding version 1.7.21 (2015-08-05 00:00:00+00:00) to apache-subversion
adding version 1.8.13 (2015-03-31 00:00:00+00:00) to apache-subversion
adding version 1.7.20 (2015-03-31 00:00:00+00:00) to apache-subversion
adding version 1.8.11 (2014-12-15 00:00:00+00:00) to apache-subversion
adding version 1.7.19 (2014-12-15 00:00:00+00:00) to apache-subversion
adding version 1.8.10 (2014-08-11 00:00:00+00:00) to apache-subversion
adding version 1.7.18 (2014-08-11 00:00:00+00:00) to apache-subversion
adding version 1.7.17 (2014-05-19 00:00:00+00:00) to apache-subversion
adding version 1.8.9 (2014-05-14 00:00:00+00:00) to apache-subversion
adding version 1.7.16 (2014-02-26 00:00:00+00:00) to apache-subversion
adding version 1.8.8 (2014-02-20 00:00:00+00:00) to apache-subversion
adding version 1.8.5 (2013-11-25 00:00:00+00:00) to apache-subversion
adding version 1.7.14 (2013-11-25 00:00:00+00:00) to apache-subversion
adding version 1.8.4 (2013-10-29 00:00:00+00:00) to apache-subversion
adding version 1.8.3 (2013-08-30 00:00:00+00:00) to apache-subversion
adding version 1.7.13 (2013-08-30 00:00:00+00:00) to apache-subversion
adding version 1.8.1 (2013-07-24 00:00:00+00:00) to apache-subversion
adding version 1.7.11 (2013-07-24 00:00:00+00:00) to apache-subversion
adding version 1.8.0 (2013-06-18 00:00:00+00:00) to apache-subversion
adding version 1.7.10 (2013-05-31 00:00:00+00:00) to apache-subversion
adding version 1.6.23 (2013-05-31 00:00:00+00:00) to apache-subversion
adding version 1.7.9 (2013-04-04 00:00:00+00:00) to apache-subversion
adding version 1.6.21 (2013-04-04 00:00:00+00:00) to apache-subversion
adding version 1.6.20 (2013-01-08 00:00:00+00:00) to apache-subversion
adding version 1.7.8 (2012-12-20 00:00:00+00:00) to apache-subversion
adding version 1.7.7 (2012-10-09 00:00:00+00:00) to apache-subversion
adding version 1.6.19 (2012-09-21 00:00:00+00:00) to apache-subversion
adding version 1.7.6 (2012-08-15 00:00:00+00:00) to apache-subversion
adding version 1.7.5 (2012-05-17 00:00:00+00:00) to apache-subversion
adding version 1.7.4 (2012-03-08 00:00:00+00:00) to apache-subversion
adding version 1.7.3 (2012-02-13 00:00:00+00:00) to apache-subversion
adding version 1.7.2 (2011-12-05 00:00:00+00:00) to apache-subversion
adding version 1.7.1 (2011-10-23 00:00:00+00:00) to apache-subversion
adding version 1.7.0 (2011-10-11 00:00:00+00:00) to apache-subversion
adding version 1.6.18 (2012-04-12 00:00:00+00:00) to apache-subversion
adding version 1.6.17 (2011-06-01 00:00:00+00:00) to apache-subversion
adding version 1.6.16 (2011-03-03 00:00:00+00:00) to apache-subversion
adding version 1.5.9 (2010-12-22 00:00:00+00:00) to apache-subversion
adding version 1.6.15 (2010-11-24 00:00:00+00:00) to apache-subversion
adding version 1.6.13 (2010-10-01 00:00:00+00:00) to apache-subversion
adding version 1.6.12 (2010-06-21 00:00:00+00:00) to apache-subversion
adding version 1.6.11 (2010-04-16 00:00:00+00:00) to apache-subversion
adding version 1.6.9 (2010-01-25 00:00:00+00:00) to apache-subversion
adding version 1.6.6 (2009-10-22 00:00:00+00:00) to apache-subversion
adding version 1.6.5 (2009-08-22 00:00:00+00:00) to apache-subversion
adding version 1.6.4 (2009-08-06 00:00:00+00:00) to apache-subversion
adding version 1.6.3 (2009-06-22 00:00:00+00:00) to apache-subversion
adding version 1.6.2 (2009-05-11 00:00:00+00:00) to apache-subversion
adding version 1.6.1 (2009-04-09 00:00:00+00:00) to apache-subversion
adding version 1.6.0 (2009-03-20 00:00:00+00:00) to apache-subversion
adding version 1.5.7 (2009-08-06 00:00:00+00:00) to apache-subversion
adding version 1.5.6 (2009-02-26 00:00:00+00:00) to apache-subversion
adding version 1.5.5 (2008-12-22 00:00:00+00:00) to apache-subversion
adding version 1.5.4 (2008-10-24 00:00:00+00:00) to apache-subversion
adding version 1.5.3 (2008-10-10 00:00:00+00:00) to apache-subversion
adding version 1.5.2 (2008-08-30 00:00:00+00:00) to apache-subversion
adding version 1.5.1 (2008-07-26 00:00:00+00:00) to apache-subversion
adding version 1.5.0 (2008-06-19 00:00:00+00:00) to apache-subversion
adding version 1.4.6 (2007-12-20 00:00:00+00:00) to apache-subversion
adding version 1.4.5 (2007-08-28 00:00:00+00:00) to apache-subversion
adding version 1.4.4 (2007-06-08 00:00:00+00:00) to apache-subversion
adding version 1.4.3 (2007-01-25 00:00:00+00:00) to apache-subversion
adding version 1.4.2 (2006-11-07 00:00:00+00:00) to apache-subversion
adding version 1.4.0 (2006-09-10 00:00:00+00:00) to apache-subversion
adding version 1.3.2 (2006-05-31 00:00:00+00:00) to apache-subversion
adding version 1.3.1 (2006-04-03 00:00:00+00:00) to apache-subversion
adding version 1.3.0 (2006-01-03 00:00:00+00:00) to apache-subversion
adding version 1.2.3 (2005-08-25 00:00:00+00:00) to apache-subversion
adding version 1.2.1 (2005-07-05 00:00:00+00:00) to apache-subversion
adding version 1.2.0 (2005-05-23 00:00:00+00:00) to apache-subversion
adding version 1.1.4 (2005-04-01 00:00:00+00:00) to apache-subversion
adding version 1.1.3 (2005-01-14 00:00:00+00:00) to apache-subversion
adding version 1.1.2 (2004-12-20 00:00:00+00:00) to apache-subversion
adding version 1.1.1 (2004-10-22 00:00:00+00:00) to apache-subversion
adding version 1.0.9 (2004-10-13 00:00:00+00:00) to apache-subversion
adding version 1.1.0 (2004-09-29 00:00:00+00:00) to apache-subversion
adding version 1.0.8 (2004-09-22 00:00:00+00:00) to apache-subversion
adding version 1.0.7 (2004-09-17 00:00:00+00:00) to apache-subversion
adding version 1.0.6 (2004-07-19 00:00:00+00:00) to apache-subversion
adding version 1.0.5 (2004-06-10 00:00:00+00:00) to apache-subversion
adding version 1.0.4 (2004-05-21 00:00:00+00:00) to apache-subversion
adding version 1.0.3 (2004-05-19 00:00:00+00:00) to apache-subversion
adding version 1.0.2 (2004-04-19 00:00:00+00:00) to apache-subversion
adding version 1.0.1 (2004-03-12 00:00:00+00:00) to apache-subversion
adding version 1.0.0 (2004-02-23 00:00:00+00:00) to apache-subversion
Skipping Subversion 1.0.0-beta1, does not match version regex
Skipping Subversion 0.37 (1.0.0-RC1), does not match version regex
Skipping Subversion 0.36, does not match version regex
...
```

I refactored a bit the script to simplify / add a few logs so it is more simple to maintain and LGTM.

**Comment by romainx** [comment]
@marcwrobel thanks for th refactoring, looks better like that 👍. Noted for the next time 😉

**Pull request merged**